docs/beta/plugins/discord.md

```diff
@@ -1,5 +1,248 @@
+
+[INSERT SCREENSHOT MESSAGE]
+
+- **Channel**:
+
+A Discord channel, e.g. [#start-here](https://discord.gg/qadM9u) {SB: link needs checking. Got link by clicking 'Create Invite' on the channel, but getting 'invalid link' msg for some reason when I try it locally). A channel node will be connected to all messages in that channel. 
+
+Found this in the code and pasting here in case the plugin exposes these channel types. 
+
+```
+ * Discord Channels can be of various types, defined below.
+ * See: https://discordapp.com/developers/docs/resources/channel#channel-object-channel-types
+ */
+export type ChannelType =
+  | "GUILD_TEXT"
+  | "DM"
+  | "GUILD_VOICE"
+  | "GROUP_DM"
+  | "GUILD_CATEGORY"
+  | "GUILD_NEWS"
+  | "GUILD_STORE";
+```
+
+- **Reaction**:
+
+A Discord emoji reaction, e.g. [insert cute emoji reaction code]. When a user reacts to a message, a reaction node is created that is connected to the author of the reaction (user that reacted) and the message reacted to. 
+
+
+[INSERT SCREENSHOT REACTIONS TO MESSAGE]
+
+
+### Edges
+
+The Discord plugin creates the following kinds of edges (connections between nodes) 
+in the contribution graph:
+
+- **Authors**:
+
+An "authors" edge connects an author (user or bot) to a message they created. 
+
+- **References**: 
+
+{SB: below is just copied and pasted from teh Discourse draft, as I'm not sure references count in discord}
+
+```
+A references edge connects a topic or post to another referencable node (i.e. a node that corresponds to a specific url on Discourse ). {SB: taken from GitHub plugin, still correct? Just say referenceable nodes are topics/posts/users?}
+
+If the reference is pointing to a user, we call it a "mention", but from SourceCred's perspective it's the same kind of edge. {SB: taken from GitHub plugin, still correct?}
+
+References in Discourse can be either hyperlinks to referencable nodes, or quoted parts of other posts {SB: I'm kind of guessing here, would be good to get more detailed explanation, and perhaps include a illustrative example (e.g. a quote that links to both text and image if supported?). This brings up another questions I've had for a while: when you select and click the "Quote" button in the UI, it automatically creates a reply, presumably creating both a reply edge and reference edge. But what if you've already started a reply, and are selecting quotes from the UI from a post different than the one you started replying to? What if you're quoting multiple posts in the same reply?}
+```
+
+- **Reply to**:
+
+{SB: below is just copied and pasted from teh Discourse draft, as I'm not sure references count in discord}
+
+```
+A reply-to edge connects a post to the post it is replying to.  
+
+
+The Discourse plugin creates the following kinds of edges (connections between nodes) 
+in the contribution graph:
+```
+- **Creates Reaction**:
```

Comment:
There's two here technically, "addsReaction" and "reactsTo".

"Adds reaction" is an edge from the person who reacted to the reaction

"Reacts to" is an edge from that reaction to the message its on.

![image](https://user-images.githubusercontent.com/7143583/91104620-5b6be000-e62b-11ea-92fa-a54575eac020.png)

Reply:
I think I got it. How's this?

- **Reacts to**:

A "Reacts to" edge connects a reaction (emoji) to the message reacted to. 

- **addsReaction**:

A "Adds Reaction" edge connects a reaction (emoji) to the author (user) of the reaction.